### DISCUSSION TITLE
Bad connection

### DISCUSSION THREAD

**Discussion by Tipmanfrank** [Hieromon/AutoConnect#578] [Q&A]
I have been testing again. I have a lot of time outs on ping and the http://IP/_ac or http://IP is answering very slowly.

I don't understand what happens. 
It seems like it is sometimes running ok.

<
#include <WiFi.h>         // Replace with WiFi.h for ESP32
#include <WebServer.h>    // Replace with WebServer.h for ESP32
#include <AutoConnect.h>  // For AutoConnect Wifi
// INPUTS
int START_ALONE = 15;  // Input to start stand alone
int START_DATA = 16;   // Input to start with homepage data
int START_NO_WIFI = 18;   // Input to start with homepage data
int BUTTON_START_ALONE = LOW;
int BUTTON_START_DATA = LOW;
int BUTTON_NO_WIFI = LOW;
int WIFICONN = LOW;

// AUTOCONNECT
WebServer Server;  // Replace with WebServer for ESP32
AutoConnect Portal(Server);
AutoConnectConfig Config;

// Webside fra ESP
void rootPage() {
  char content[] = "Kapsejlads.nu - HORN<br>Afsted med dem!";
  Server.send(200, "text/plain", content);
}

void setup() {
  Serial.begin(115200);
  Serial.println("SETUP START");

  // Input Output
  // pinMode(START_ALONE, INPUT);
  // pinMode(START_DATA, INPUT);
  // pinMode(START_NO_WIFI, INPUT);
  Serial.println("I/O OK");

  // AUTOCONNECT
  Config.apid = "Kapsejlads-horn";
  Config.psk = "Kapsejlads.nu";
  Portal.config(Config);
  Serial.println("AUTO CONNECT OK");

  BUTTON_START_DATA = digitalRead(START_DATA);
  Serial.print("  BUTTON_START_DATA = ");
  Serial.println(BUTTON_START_DATA);
  // if(BUTTON_START_DATA==HIGH){
    //SHOW AUTOCONNECT SCREEN //(OFF with button during startup)
    Server.on("/", rootPage);
    if (Portal.begin()) { //SWITCH ON NETWORK MAN SCREEN
      Serial.println("  WiFi connected: " + WiFi.localIP().toString());
    }
    Serial.println("ROOTPAGE OK");
  // }
  Serial.println("BUTTON WIFI START OK");

}

void loop() {
  // put your main code here, to run repeatedly:
  Portal.handleClient();
  Serial.println(millis());

}
>

**Comment by Hieromon** [comment]
I cannot reproduce the problem in my environment. 

Generally, when extreme response delays occur with such a mundane and basic sketch, the following conditions should be checked:

1. Sufficient power and current
A common failure is an insufficient current. Current consumption increases, especially when SoftAP is activated simultaneously. Most importantly, the voltage drop on the VBUS (5V) line due to inrush current must be less than 330mV. Can the power supply lines connected to your ESP32 module (especially the LDO on the ESP module) withstand that?

2. LAN segment configuration with WiFi access points
Are there any obstacles in the path of your network with your WiFi access points? Does your WiFi network broadcast on different channels with the same SSID? Are you able to isolate them?

Then, **Isolate AutoConnect** to narrow down the cause of the problem and **verify it with some example sketches provided by the ESP32 WebServer library**.

Then, if AutoConnect behavior is still not stable, please refer to the documentation [here](https://hieromon.github.io/AutoConnect/faq.html#4-reports-the-issue-to-autoconnect-github-repository) and provide the information I need to investigate.

- Build with `Core Debug Level=Verbose` in the Arduino IDE and investigate any logs that may be contributing to the delay.
- Enable the `AC_DEBUG` macro to get AutoConnect traces. Investigate error factors as well as core debug logs. See the Documentation [here](https://hieromon.github.io/AutoConnect/faq.html#3-turn-on-the-debug-log-options).

Until I find out that the cause lies in AutoConnect, I will temporarily move to Discussions.

**Reply by Tipmanfrank** [reply]
I am very happy for your support. And I love AutoConnect so much over WifiManager since I can change network after ESP has connected to a network.

I have both 2.4 and 5 GHz network with same SSID. So maybe it tries to connect to the 5GHz now and then. I was really frustrated because sometimes it was running fine, other times it was really bad.

I tried to connect the ESP to my iphone and then I can ping it all the time and the connection seems to be stable. So I think that the mix of 2.4 and 5 GHz network made the problems. That will be a problem at home, but the device will normally only be used outside my home. So I can live with the problems at home. But I have spend half a day to figure out what was wrong.

Since the connection to the phone solved the problem I didn't focus anymore on the supply voltage. It has been supplied over USB from a PC and a power bank. In the finally setup it will be supplied from a 12 V car battery with an LM7805 component to make 5 V DC. This gives a very stable 5 V everytime I have measured it.

**Comment by Hieromon** [comment]
> I have both 2.4 and 5 GHz network with same SSID.

?? Does your WiFi environment assume the band-steering? Of course the iPhone will connect, since iOS supports 802.11k roaming. But I am not certain where that support currently stands in ESP-IDF. Was this supported in the Arduino core?
Refs: https://github.com/espressif/esp-idf/issues/3671

> In the finally setup it will be supplied from a 12 V car battery with an LM7805 component to make 5 V DC.

What is important is not the rating, but whether the front-stage LDO can withstand the ESP32's sink current. The LM7805 isn't so responsive. In my experiments, the 5V line momentarily drops nearly 600mV, dragged down by the 3.3V line's rapid current sink. It depends on the circuit impedance, so I won't say it explicitly, but I personally don't recommend relying on the LM7805 alone.

Your ESP module probably uses the AMS1117 series, and this chip doesn't have the headroom. Using the ADP3338 in front of the LM7805 ensures a stable power line even if the ESP32 eats a lot of food. But, in the end, you can do whatever you like, so just for your reference.

**Reply by Tipmanfrank** [reply]
Band steering? I am not sure what to answer here. I use Unifi Dream Machine - more or less standard setup.

I will investigate further about ADP3338.

**Reply by Hieromon** [reply]
Band steering steers Wi-Fi clients that support 5 GHz to the 5 GHz network. This reduces congestion on the 2.4 GHz network for Wi-Fi clients. When the 5 GHz network is congested, the least active Wi-Fi client will be moved to his 2.4 GHz network. 

In this case, if Band Steering is enabled with Dream Machine, ESP32 which does not implement IEEE802.11k/r/v will not roam successfully and hang up in the worst case. So, the devices do actually connect to the 5Ghz network whenever they can, but if they stay on the slower network, it may be worth it to separate the two.

Below is a quote from [UniFi Dream Machine datasheet](https://dl.ui.com/ds/udm_ds).

> Advanced RF Performance
> RF performance and configuration features include spectral analysis, airtime fairness, band steering, and cell-size tuning.

**Reply by Tipmanfrank** [reply]
Excellent answer. I have now made an additional wifi network in my Dream Machine which only has 2.4 GHz network available. So all my IOT devices can be connected to this network. Then they will not try to connect to the 5 GHz network anymore.

**Comment by Tipmanfrank** [comment]
I am still having some problems.

I have this Dream Machine. I have setup a Wifi called "Slottet" with mixed 5 GHZ and 2.4 GHz. I have also made another Wifi called "Slottet IOT" only with 2.4 GHz. I the connect ESP32 to "Slottet IOT". But after a restart of the ESP then it connects itself to "Slottet" again. Why? 

I thought it could only connect to the last connected wifi.

**Reply by Tipmanfrank** [reply]
I actually solved it. With esptool I erased all data and all known SSIDs. Now it only connects to "Slottet IOT". But it is a hell of a job to use esptool.
